FeedStoreChallenge/CoreDataFeedStore.swift

```diff
@@ -26,17 +28,63 @@ public final class CoreDataFeedStore: FeedStore {
 			url: storeURL
 		)
 		context = container.newBackgroundContext()
+		cacheStore = ManagedCacheStore(context: context)
+		feedImageStore = ManagedFeedImageStore(context: context)
 	}
 
 	public func retrieve(completion: @escaping RetrievalCompletion) {
-		fatalError("Must be implemented")
+		context.perform { [cacheStore] in
+			do {
+				if let cache = try cacheStore.cache() {
+					let localFeedImages = cache.feedImages.map { $0.localFeedImage }
+					completion(.found(feed: localFeedImages, timestamp: cache.timestamp))
```

Comment:
It's possible to extract this mapping logic to one of the helpers, for example following the same approach of `ManagedFeedImage`'s extension with a `localFeedImage`, it's possible to implement a `ManagedCache` `localFeed` computed property, that returns `[LocalFeedImage]`:
```suggestion
					completion(.found(feed: cache.localFeed, timestamp: cache.timestamp))
```